File: spring-cloud-contract/java-spring-cloud-contract/app-caller/src/main/java/com/example/javaspringcloudcontract/service/CallService.java
package com.example.javaspringcloudcontract.service;

import com.example.javaspringcloudcontract.external.CalleeCaller;
import com.example.javaspringcloudcontract.presentation.ResponseBody;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class CallService {
    private final CalleeCaller calleeCaller;

    public ResponseBody callData(int num) {

        ResponseEntity<String> calleeResult = calleeCaller.callCalleeApplication(num);

        return new ResponseBody(calleeResult.getStatusCode().value(), calleeResult.getBody());
    }
}

File: spring-cloud-contract/kotlin-spring-cloud-contract/app-caller/src/main/kotlin/com/example/springcloudcontract/service/CallService.kt
package com.example.springcloudcontract.service

import com.example.springcloudcontract.external.CalleeCaller
import com.example.springcloudcontract.presentation.ResponseBody
import org.springframework.stereotype.Service

@Service
class CallService(
    private val calleeCaller: CalleeCaller,
) {
    fun callData(num: Int): ResponseBody {
        val calleeResult = calleeCaller.callCalleeApplication(num)

        return ResponseBody(calleeResult.statusCode.value(), calleeResult.body!!)
    }
}